defs.bzl

```diff
@@ -154,6 +154,22 @@ def _parse_artifact_str(artifact_str):
     else:
         return parse.parse_maven_coordinate(artifact_str)
 
+JAVA_EXPORT_REPO_NAME = "internal_java_exports_maven"
+
+def java_export_deps():
+    """
+        Adds the maven dependencies that are needed to run java_export rule.
+    """
+    maven_install(
```

Comment:
Any reason not to use a `jvm_import` to pull in a jar we add to the repo? Other than the vast number of dependencies? 

The current way of using `rules_jvm_external` doesn't need anyone to go through the cumbersome "import the ruleset, load the rule set deps, run the deps setup" dance. That's why the current deployer doesn't use anything outside the Java 8 JRE.

Reply:
Maybe I'm wrong, but AFAIK bazel never supported transitive deps, so the only real way to avoid forcing the user from adding those deps is to save the jars in our repo, which isn't a solution IMHO.
Am I wrong about it? Even with jvm_import it will include target from different repo, which the user will need to add.

Reply:
If it's just a single (or few) jar dependency, we could vendor it.

How about doing it this way instead of requiring a load statement? https://github.com/bazelbuild/rules_jvm_external#exporting-and-consuming-artifacts-from-external-repositories

Reply:
The GCS lib has 23 deps (so 24 total jars), we can vendor them but I'm not sure it's the right approach.
LMK what you think about it, if you prefer it to be vendored I'll change it.

Reply:
@shs96c and I had a discussion offline about this, and I think that it's would be a cleaner approach to go with your implementation, with an internal `maven_install` repo.

Reply:
:+1: